sympy/integrals/tests/test_heurisch.py

```diff
@@ -71,7 +71,9 @@ def test_heurisch_exp():
     assert heurisch(Integral(x**z*y, (y, 1, 2), (z, 2, 3)).function, x) == (x*x**z*y)/(z+1)
     assert heurisch(Sum(x**z, (z, 1, 2)).function, z) == x**z/log(x)
 
-
+def test_issue_10680():
+    assert integrate(x**log(x**log(x**log(x))),x)
```

Comment:
This could be

``assert isinstance(integrate(x**log(x**log(x**log(x))),x), Integral)``

to show that no error occurs and an unevaluated integral is returned.

You can edit this and then commit and push it to github. It will be automatically added to the pull request. (Run tests again before pushing.)